File: C#/11047.cs
using System;

class Program
{
    static void Main()
    {
        int answer = 0;
        string[] input = Console.ReadLine().Split(' ');
        int coins = int.Parse(input[0]);
        int goal = int.Parse(input[1]);
        int[] coinsValue = new int[coins];

        for (int idx = 0; idx < coins; idx++)
        {
            coinsValue[idx] = int.Parse(Console.ReadLine());
        }

        Array.Reverse(coinsValue);

        while (goal != 0)
        {
            for (int idx = 0; idx < coinsValue.Length; idx++)
            {
                if (coinsValue[idx] <= goal)
                {
                    goal -= coinsValue[idx];
                    answer++;
                    idx--;
                }
            }
        }
        Console.WriteLine(answer);
    }
}

File: C++/11047.cpp
#include <iostream>
#include <vector>    
#include <algorithm> 

using namespace std;

int main()
{
    int answer = 0;
    int coins, goal;
    cin >> coins >> goal;

    vector<int> coinsValue(coins);

    for (int idx = 0; idx < coinsValue.size(); idx++)
    {
        cin >> coinsValue[idx];
    }

    reverse(coinsValue.begin(), coinsValue.end());

    while (goal != 0)
    {
        for (int idx = 0; idx < coinsValue.size(); idx++)
        {
            if (goal >= coinsValue[idx])
            {
                goal -= coinsValue[idx];
                idx--;
                answer++;
            }
        }
    }
    cout << answer;
    return 0;
}